In [1]:
# import the libraries
import pandas as pd
import numpy as np
import statistics

In [2]:
# Reading dataset to python
data=pd.read_csv('StudentsPerformance.csv')
data


gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

# 1. Find out how many males and females participated in the test.


In [3]:
data['gender'].value_counts()


female    518
male      482
Name: gender, dtype: int64

518 females and 482 males participated in the test

# 2. What do you think about the students' parental level of education?

In [4]:
data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

# 3. Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course

In [7]:
# Average based on gender
avggen = data[['gender', 'math score', 'reading score', 'writing score']].groupby('gender').mean()
avggen

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

Female students have higher average for reading and writing score while male students have higher average in maths score..

In [8]:
# Average based on test preparation course
avgtest= data[['test preparation course', 'math score', 'reading score', 'writing score']].groupby('test preparation course').mean()
avgtest

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

Students who completed test preparation course have higher average score for all scores

# 4. What do you think about the scoring variation for math, reading and writing based on
● Gender
● Test preparation course

In [9]:
# Variation based on gender 
vargen= data[['gender', 'math score', 'reading score', 'writing score']].groupby('gender').std()
vargen

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

Male students have less variations for each of the scores.

In [10]:
# Variation based on test preparation course
vartest= data[['test preparation course', 'math score', 'reading score', 'writing score']].groupby('test preparation course').std()
vartest

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

The variation for those who completed test preparation course is less compared to those who not done .

# 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.

In [11]:
# Top 25% students based on maths score
top_students = data.sort_values('math score', ascending=False).nlargest(250,"math score")
top_students

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
856    male        group E            some high school      standard   
855    male        group E          associate's degree  free/reduced   
854    male        group D            some high school      standard   
853    male        group D          associate's degree  free/reduced   
852    male        group D                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
856               completed          77             76             77  
855               completed          77             69             68  
854               completed          77             68             69  
853                    none          77             78             73  
852                    none          77             62             64  

[250 rows x 8 columns]